### DISCUSSION TITLE
upstream request/response logs

### DISCUSSION THREAD

**Discussion by ahmadnassri** [Kong/kong#7216] [Ideas and feature requests]
just like any of the current logging plugins, but to log the actual upstream request and response sent between Kong and the upstream server.

this is useful for debugging upstream requests and identifying issues between Kong and the upstream server, regardless of what is sent back tot he client side.

whether this can be a separate plugin or part of the existing plugins is up for debate.

**Comment by sonicaghi** [comment]
:+1:

**Comment by ahmadnassri** [comment]
another use for this is to see exactly how the request is formed AFTER a transformation and how a response is received BEFORE a transformation

**Comment by jsettypalli** [comment]
Is this feature released?

**Comment by jmgpeeters** [comment]
This would be a great feature.

**Comment by cocktailkrabbe** [comment]
Need this to inspect "An invalid response was received from the upstream server" (Status Code 502)

**Comment by somada141** [comment]
This is a long shot since this has been open for 3 years but I thought I'd give it a shot since I'm out of ideas. Is there a way to inspect what Kong sends to the upstream server so that I can debug issues with the `request-transformer` plugin?

As the communication happens over SSL and I don't want to install MITM stuff on my own production server I was hoping for a 'cleaner' solution.

**Comment by janv8000** [comment]
> This is a long shot since this has been open for 3 years but I thought I'd give it a shot since I'm out of ideas. Is there a way to inspect what Kong sends to the upstream server so that I can debug issues with the `request-transformer` plugin?
 
You could consider using [Mockbin by Kong](https://mockbin.org/docs#echo)

**Comment by somada141** [comment]
Actually that makes total sense, dunno why I didn't think of that. Thanks @janv8000 I'll give it a go!

**Comment by justlife787** [comment]
> > This is a long shot since this has been open for 3 years but I thought I'd give it a shot since I'm out of ideas. Is there a way to inspect what Kong sends to the upstream server so that I can debug issues with the `request-transformer` plugin?
> 
> You could consider using [Mockbin by Kong](https://mockbin.org/docs#echo)

but how can you use mock bin within-firm boundary, this would be security concerns

**Comment by janv8000** [comment]
> but how can you use mock bin within-firm boundary, this would be security concerns

Install and run your own instance: https://github.com/Kong/mockbin#running
